### PR TITLE
fix: wired position cursor jump when Xcode trim whitespace preference is true.

### PR DESCRIPTION
**Pull request by r-plus**
FIx https://github.com/XVimProject/XVim2/issues/351
FIx https://github.com/XVimProject/XVim2/issues/296

testsed `i`, `I`, `j` and `o` operations.

### CODE REVIEW

File: XVim2/Xcode/SourceEditorViewProxy+XVim.m [hunk 1]
```diff
@@ -131,8 +131,19 @@ - (void)xvim_endEditTransaction
     }
 }
 
+/// XVim last set insertionPoint is wrong when Xcode trim whitespace.
+/// If insertionPoint is wrong, wired position cursor jump will occur when `j` and `o` operation.
+/// This method fixing insertionPoint for that operation.
+- (void)xvim_fixInsertionPoint
+{
+    if (self.selectionMode == XVIM_VISUAL_NONE && self.insertionPoint > self.selectedRange.location) {
```
Comment (r-plus) [inline XVim2/Xcode/SourceEditorViewProxy+XVim.m:139]: if selectionMode condition is nothing, `j` will not move cursor in visual line mode when current cursor is blank line.

**Comment by pebble8888** [comment]
I think the problem is not fixed.

How to reproduce:
macOS11.2.3
Xcode12.4

スクリーンショット 2021-04-18 6 23 01

  
`4ljo<ESC>i<SPACE><SPACE><SPACE><SPACE><ESC>j` in `File.swift`  
  

スクリーンショット 2021-04-18 6 21 46
  
And `i`  
  
スクリーンショット 2021-04-18 6 22 33

**Comment by r-plus** [comment]
ah, thanks.
looks like different behavior on release build and debug build.

**Comment by r-plus** [comment]
wait, I'm testing #373 branch build.
no problem when I tried it again on this branch build.

@pebble8888 Any other condition to show problem?


https://user-images.githubusercontent.com/425216/115133074-5d623380-a040-11eb-9f74-fd5fab0b6c66.mov

**Comment by r-plus** [comment]
oh, sorry last operation is `i`
i just type `l`.

confirmed problem.

**Comment by r-plus** [comment]
ok, not yet covered `i` operation.
I'll add workaround for that operation.

**Comment by r-plus** [comment]
fixed for all of the normal evaluator.

**Comment by pebble8888** [comment]
Thank you.
Great improvement.

**Pull request merged**